tensorflow/core/kernels/resize_bilinear_op_gpu.cu.cc

```diff
@@ -25,12 +25,78 @@ limitations under the License.
 #include "tensorflow/core/platform/types.h"
 #include "tensorflow/core/util/gpu_kernel_helper.h"
 
+// auxilary 16-byte datatype for ResizeBilinearKernel_faster
+// the fields are not important. The only purpose of this is to read 16 bytes
+// from GPU gloal memory
+struct four_floats{
+  float a;
+  float b;
+  float c;
+  float d;
+};
+
 namespace tensorflow {
 
 typedef Eigen::GpuDevice GPUDevice;
 
 namespace {
 
+template <typename T, int C_UNROLL>
```

Comment:
If you are not planning to implement this kernel for anything but float, you should remove the template parameter T. If you are planning to support other types as well, add a static_assert for now.

It also looks like you use a fixed number for C_UNROLL. I would just use a constexpr in the implementation.

Reply:
so instead of the template argument `C_UNROLL`, you are recommending using `constexpr c_unroll = ...`? Sure, but just out of curiosity, is there a reason you prefer the latter over the former?

Reply:
A template argument which is instantiated for a single value seems unnecessary.

Yes: half, float, and double are the most interesting ones.

Reply:
> A template argument which is instantiated for a single value seems unnecessary.

Okay, will use `const expr c_unroll = ...` over the template.

> If you are planning to support other types as well, add a static_assert for now.
shouldn't we use regular `assert` over `static_assert` since we want the assert to evaluate at runtime not at compile time? Let me know what you think since I'm not an expert on asserts.

Reply:
@chsigg ping!